File: introduction/calculator.py
#Inputs - accept data from the user
#User will choose the desired operation
#Operation - perform the operation
#Output - display the result to the user


def calculator():
     print("Welcome to the calculator.")
     print("Select operation: ")

     print("1. Addition (+)")
     print("2. Subtraction (-)")
     print("3. Multiplication (*)")
     print("4. Division (/)")

     choice =  input("Enter choice (1/2/3/4): ") #variable

     if choice in ('1', '2', '3', '4'):
          num1 = float(input("Enter first number: "))
          num2 = float(input("Enter second number: "))

          if choice == '1':
               result = num1 + num2
               operation = 'Addition'
          elif choice == '2':
               result = num1 - num2
               operation = 'Subtraction'
          elif choice == '3':
               result = num1 * num2
               operation = 'Multiplication'
          elif choice == '4':
               if num2 == 0:
                    print("Error: Division by zero")
                    return
               result = num1 / num2
               operation = 'Division'
          print(f"{operation} of {num1} and {num2} is {result}")
     else:
          print("Select a valid operation.")


#Run the calculator function
calculator()

# PS D:\Python-Classes\introduction> python3 calculator.py
# Welcome to the calculator.
# Select operation: 
# 1. Addition (+)
# 2. Subtraction (-)
# 3. Multiplication (*)
# 4. Division (/)
# Enter choice (1/2/3/4): 1
# Enter first number: 30
# Enter second number: 45
# Addition of 30.0 and 45.0 is 75.0
# PS D:\Python-Classes\introduction> 

File: introduction/calculator/calculator-frontend/src/components/calculator.js
import React, { useState } from "react";

const Calculator = () => {
    const [num1, setNum1] = useState("");
    const [num2, setNum2] = useState("");
    const [operation, setOperation] = useState("addition");
    const [result, setResult] = useState(null);
    const [error, setError] = useState(null);

    const calculate = async () => {
        setError(null);
        setResult(null);

        try {
            const response = await fetch("http://127.0.0.1:8000/api/calculate/", {
                method: "POST",
                headers: { "Content-Type": "application/json" },
                body: JSON.stringify({ num1, num2, operation }),
            });

            const data = await response.json();
            if (response.ok) {
                setResult(data.result);
            } else {
                setError(data.error);
            }
        } catch (error) {
            setError("Failed to connect to server.");
        }
    };

    return (
        <div>
            <h2>Calculator</h2>
            <input type="number" value={num1} onChange={(e) => setNum1(e.target.value)} placeholder="First number" />
            <input type="number" value={num2} onChange={(e) => setNum2(e.target.value)} placeholder="Second number" />
            <select value={operation} onChange={(e) => setOperation(e.target.value)}>
                <option value="addition">Addition</option>
                <option value="subtraction">Subtraction</option>
                <option value="multiplication">Multiplication</option>
                <option value="division">Division</option>
            </select>
            <button onClick={calculate}>Calculate</button>

            {result !== null && <h3>Result: {result}</h3>}
            {error && <p style={{ color: "red" }}>{error}</p>}
        </div>
    );
};

export default Calculator;
